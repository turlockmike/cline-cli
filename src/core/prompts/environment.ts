import * as os from 'os';

export function getEnvironmentInfo() {
    return `
Environment Information:
Operating System: ${os.platform()} ${os.release()}
Architecture: ${os.arch()}
CPU Cores: ${os.cpus().length}
Total Memory: ${Math.round(os.totalmem() / (1024 * 1024 * 1024))}GB
Free Memory: ${Math.round(os.freemem() / (1024 * 1024 * 1024))}GB
Default Shell: ${process.env.SHELL || 'unknown'}
Home Directory: ${os.homedir()}
Current Working Directory: ${process.cwd()}
Node Version: ${process.version}
Current Time: ${new Date().toLocaleString()}
Locale: ${Intl.DateTimeFormat().resolvedOptions().locale}
Timezone: ${Intl.DateTimeFormat().resolvedOptions().timeZone}
User Info: ${os.userInfo().username}
Package Manager: ${process.env.npm_config_user_agent?.split('/')[0] || 'npm'}
Terminal: ${process.env.TERM_PROGRAM || process.env.TERM || 'unknown'}
Git Branch: ${require('child_process').execSync('git rev-parse --abbrev-ref HEAD 2>/dev/null || echo "not a git repo"').toString().trim()}
`;
} 