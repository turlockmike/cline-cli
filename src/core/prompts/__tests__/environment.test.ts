import { getFilesInCurrentDirectory } from '../environment';
import { readdirSync } from 'node:fs';

// Mock the fs module
jest.mock('node:fs', () => ({
    readdirSync: jest.fn()
}));

describe('getFilesInCurrentDirectory', () => {
    it('returns max/3 files from start, middle, and end with ellipsis', () => {
        const mockFiles = ['1.ts', '2.ts', '3.ts', '4.ts', '5.ts', '6.ts', '7.ts', '8.ts', '9.ts', '10.ts',
                          '11.ts', '12.ts', '13.ts', '14.ts', '15.ts', '16.ts', '17.ts', '18.ts', '19.ts', '20.ts'];
        (readdirSync as jest.Mock).mockReturnValue(mockFiles);
        
        expect(getFilesInCurrentDirectory(9)).toEqual([
            '1.ts', '2.ts', '3.ts',
            '...',
            '10.ts', '11.ts', '12.ts',
            '...',
            '18.ts', '19.ts', '20.ts'
        ]);
    });
}); 