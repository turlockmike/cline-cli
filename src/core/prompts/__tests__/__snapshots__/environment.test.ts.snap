// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment Tests getEnvironmentInfo returns consistent environment information 1`] = `
"
    <environment_details>
Environment Information:
Operating System: linux 6.8.0-1021-azure
Architecture: x64
CPU Cores: 4
Total Memory: 16GB
Free Memory: 14GB
Default Shell: /bin/bash
Home Directory: /home/runner
Current Working Directory: /home/runner/work/hataraku/hataraku
Files in Current Directory: .changeset,.cursor,.eslintrc.json,.git,.gitattributes,.github,.gitignore,.husky,.npmrc,.nvmrc,.prettierignore,.prettierrc.json,CHANGELOG.md,CONTRIBUTING.md,LICENSE,README.md,docs,esbuild.js,jest.config.cjs,node_modules,package-lock.json,package.json,src,tsconfig.cjs.json,tsconfig.esm.json,tsconfig.json
Node Version: v20.6.0
Current Time: 2/19/2025, 2:18:28 PM
Locale: en-US
Timezone: UTC
User Info: runner
Package Manager: npm
Terminal: undefined
Git Branch: HEAD
</environment_details>
"
`;
